### PR TITLE
cover contain

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: lib/resize.js [hunk 1]
```diff
@@ -2,9 +2,21 @@ const Jimp = require('jimp');
 const TinyColor = require('tinycolor2');
 
 module.exports = async (options, buffer) => {
-  const { width, height, background } = options;
+  const { background, type } = options;
+  let { width, height } = options;
+  // if only one of width/height was provided, make it a square:
+  if (width && !height) {
```
Comment (jgallen23) [inline lib/resize.js:8]: if you only provide one, then it should auto decide the other dimension - that should be part of jimp, right?

File: lib/resize.js [hunk 2]
```diff
@@ -2,9 +2,21 @@ const Jimp = require('jimp');
...
+    height = width;
+  }
+  if (height && !width) {
+    width = height;
+  }
   const jimpImage = await Jimp.read(buffer);
-  jimpImage.resize(width, height);
+  if (type) {
```
Comment (jgallen23) [inline lib/resize.js:15]: can we just default type to resize?

File: lib/resize.js [hunk 3]
```diff
@@ -2,9 +2,21 @@ const Jimp = require('jimp');
 const TinyColor = require('tinycolor2');
 
 module.exports = async (options, buffer) => {
```
Comment (jgallen23) [inline lib/resize.js:4]: might be cleaner to do `({ background, type, width, height }, buffer)`. thoughts?

File: lib/resize.js (hunk 1 above)
Comment (orthagonal) [reply lib/resize.js:8]: You would think, but nope it only works if you provide both.

Comment (jgallen23) [reply lib/resize.js:6]: https://github.com/oliver-moran/jimp/tree/master/packages/plugin-resize#usage - if it doesn't exist, set to auto

Comment (jgallen23) [reply lib/resize.js:6]: https://github.com/oliver-moran/jimp/tree/master/packages/plugin-resize#usage - if one or the other is not provided, set to auto

Comment (orthagonal) [reply lib/resize.js:6]: This works for resize, but cover and contain still fail if you don't provide both.  This covers
all possible cases.

**Pull request merged**